File: apps/client/lib/features/auth/domain/repositories/auth_repository.dart
import '../entities/auth_session.dart';

/// The auth port. Infrastructure supplies the API-backed implementation.
/// Methods throw a [Failure] (see core/errors) on failure.
abstract interface class AuthRepository {
  Future<AuthSession> login({required String email, required String password});
  Future<AuthSession> register({
    required String fullName,
    required String email,
    required String password,
  });
  Future<void> logout();
  Future<void> forgotPassword(String email);

  /// Native Google sign-in → Omnisift session. Returns null if the user cancels.
  Future<AuthSession?> googleSignIn();

  /// Returns a stored session if the user is already signed in, else null.
  Future<AuthSession?> currentSession();
}

File: apps/backend/src/modules/auth/domain/repositories/auth.repository.ts
import { AuthSession } from '../entities/auth-session.entity';

/** The auth port. Infrastructure supplies the Supabase-backed implementation. */
export abstract class AuthRepository {
  abstract register(input: {
    fullName: string;
    email: string;
    password: string;
  }): Promise<AuthSession>;
  abstract login(input: { email: string; password: string }): Promise<AuthSession>;
  abstract googleSignIn(idToken: string, accessToken?: string): Promise<AuthSession>;
  abstract refresh(refreshToken: string): Promise<AuthSession>;
  abstract logout(accessToken: string): Promise<void>;
  abstract forgotPassword(email: string): Promise<void>;
}
